File: Week2 Assignment/SpiralMatrix.c
//Print matrix in spiral form

#include <stdio.h>

void printSpiral(int rows, int cols, int matrix[rows][cols]) {
    int top = 0, bottom = rows - 1;
    int left = 0, right = cols - 1;
    int i;

    while (top <= bottom && left <= right) {
        // Print top row
        for (i = left; i <= right; i++)
            printf("%d ", matrix[top][i]);
        top++;

        // Print right column
        for (i = top; i <= bottom; i++)
            printf("%d ", matrix[i][right]);
        right--;

        // Print bottom row
        if (top <= bottom) {
            for (i = right; i >= left; i--)
                printf("%d ", matrix[bottom][i]);
            bottom--;
        }

        // Print left column
        if (left <= right) {
            for (i = bottom; i >= top; i--)
                printf("%d ", matrix[i][left]);
            left++;
        }
    }
}

int main() {
    int rows, cols,i,j;
    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols];
    printf("Enter matrix elements:\n");
    for (i = 0; i < rows; i++)
        for (j = 0; j < cols; j++)
            scanf("%d", &matrix[i][j]);

    printf("Spiral Order:\n");
    printSpiral(rows, cols, matrix);

    return 0;
}

File: Week2 Assignment/SpiralMatrix.py
#Print matrix in spiral form

def print_spiral(matrix):
    top, bottom = 0, len(matrix) - 1
    left, right = 0, len(matrix[0]) - 1

    while top <= bottom and left <= right:
        # Top row
        for i in range(left, right + 1):
            print(matrix[top][i], end=' ')
        top += 1

        # Right column
        for i in range(top, bottom + 1):
            print(matrix[i][right], end=' ')
        right -= 1

        # Bottom row
        if top <= bottom:
            for i in range(right, left - 1, -1):
                print(matrix[bottom][i], end=' ')
            bottom -= 1

        # Left column
        if left <= right:
            for i in range(bottom, top - 1, -1):
                print(matrix[i][left], end=' ')
            left += 1

# Input
rows = int(input("Enter number of rows: "))
cols = int(input("Enter number of columns: "))

print("Enter matrix elements row-wise:")
matrix = []
for _ in range(rows):
    matrix.append(list(map(int, input().split())))

print("Spiral Order:")
print_spiral(matrix)
